napari/utils/events/evented_model.py

```diff
@@ -1,19 +1,32 @@
+import operator
 import warnings
 from typing import ClassVar, Dict, Set
 
 from pydantic import BaseModel, PrivateAttr
+from pydantic.main import ModelMetaclass
 
+from ...utils.misc import EqOperator, pick_equality_operator
 from .custom_types import JSON_ENCODERS
-from .dataclass import _type_to_compare, is_equal
 from .event import EmitterGroup, Event
 
 
-class EventedModel(BaseModel):
+class EqualityMetaclass(ModelMetaclass):
```

Comment:
Can this get a short doc string about how we're using the Metaclass to do the equality check (what you wrote in PR is enough)